### PR TITLE
Add keyboardAppearance prop to TextInput component.

### LINKED ISSUE #3649: [iOS] change keyboard color theme?
We are refactoring our iOS app with react-native. 
There is a screen where the background is all white and we need a numeric keyboard in dark theme - to separate the keyboard with the background.

I did't find anywhere in the doc or existing parts that allow me to do this, could anyone point me the files that are responsible for this so that I could take a look in?

### PR DESCRIPTION
**Pull request by christopherdro**
Resolves #3649.

### CODE REVIEW

File: React/Base/RCTConvert.m [hunk 1]
```diff
@@ -272,6 +272,13 @@ + (NSDate *)NSDate:(id)json
   @"numeric": @(UIKeyboardTypeDecimalPad),
 }), UIKeyboardTypeDefault, integerValue)
 
+RCT_ENUM_CONVERTER(UIKeyboardAppearance, (@{
+  @"default": @(UIKeyboardAppearanceDefault),
+  @"light": @(UIKeyboardAppearanceLight),
+  @"alert": @(UIKeyboardAppearanceAlert),
+  @"dark": @(UIKeyboardAppearanceDark),
+  }), UIKeyboardAppearanceDefault, integerValue)
```
Comment (ide) [inline React/Base/RCTConvert.m]: style nit: unindent 2 spaces

**Comment by ide** [comment]
Looks good and is consistent with the other APIs. If you have time, could you add an example to the UIExplorer too?

**Comment by nicklockwood** [comment]
As @ide says, an example would be great. Other than that, looks good to me.

**Comment by christopherdro** [comment]
@ide @nicklockwood Done.

**Comment by ide** [comment]
Thanks!

**Comment by nicklockwood** [comment]
@facebook-github-bot import

**Comment by nicklockwood** [comment]
Note: I removed the "alert" style, as it's deprecated.

**Pull request closed without merging**